packages/store/src/focus/reducer.ts

```diff
@@ -68,6 +87,38 @@ export const getFocusTree: Selector<Node | null, [string?]> = createSelector(
   }
 )
 
+/**
+ * [[Selector]] that returns the focus path from the leaf with the given id
+ *
+ * @param id - optional id of the document that should be considered as the leaf of the focus path. By default, we use the currently focused document of the current scope
+ * @returns an array of ids of the documents that are part of the focus path (i.e. the focused document and their ancestors). `null`, if there exists no focus path
+ */
+export const getFocusPath: Selector<
+  string[] | null,
+  [string?]
+> = createSelector(
+  (
+    state: ScopedState,
+    leaf: string | null = getFocused()(state)
+  ): string[] | null => {
+    if (!leaf) return null
+    const root = getFocusTree()(state)
+    if (!root) return null
+
+    let current = leaf
+    let path: string[] = [leaf]
+
+    while (current !== root.id) {
+      const parent = findParent(root, current)
+      if (!parent) return null
```

Comment:
This is a case, that can't happen due to the way the tree is build up, right? Otherwise maybe think about returning `path` anyway?

Reply:
Against that. Since this is a semantically different case to an empty path (e.g. when nothing is focused). `null` should only be returned if you provide an invalid id